evm-contracts/src/v0.7/dev/CompoundPriceFlaggingValidator.sol

```diff
@@ -0,0 +1,277 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import "./ConfirmedOwner.sol";
+import "../vendor/SafeMathChainlink.sol";
+import "../interfaces/FlagsInterface.sol";
+import "../interfaces/AggregatorV3Interface.sol";
+import "../interfaces/UniswapAnchoredView.sol";
+import "./UpkeepCompatible.sol";
+
+/**
+ * @notice This validator compares the price of Chainlink aggregators against
+ * their equivalent Compound Open Oracle feeds. For each aggregator, a Compound
+ * feed is configured with its symbol, number of decimals, and deviation threshold.
+ * An aggregator address is flagged when its corresponding Compound feed price deviates
+ * by more than the configured threshold from the aggregator price.
+ */
+contract CompoundPriceFlaggingValidator is ConfirmedOwner, UpkeepCompatible {
+  using SafeMathChainlink for uint256;
+
+  struct CompoundFeedDetails {
+    // Used to call the Compound Open Oracle
+    string symbol;
+    // Used to convert price to match aggregator decimals
+    uint8 decimals;
+    // 1        = 100%
+    // 5        = 20%
+    // 10       = 10%
+    // 20       = 5%
+    // 50       = 2%
+    // 100      = 1%
+    // 200      = 0.5%
+    // 500      = 0.2%
+    // 1000     = 0.1%
+    uint32 deviationThresholdDenominator;
```

Comment:
In other places we've added a fixed denominator(parts per billion) and then allow the numerator to be configured. I think that way is a little less likely to lose precision.

Reply:
True, although my preference here is to perform as little arithmetic as possible when validating each feed.

If we use the numerator, the validation needs an extra `mul()` to calculate the deviation threshold amount.
So:
```
uint256 deviationAmountThreshold = aggregatorPrice.div(uint256(compDetails.deviationThresholdDenominator));
```

becomes

```
uint256 deviationAmountThreshold = aggregatorPrice.mul(uint256(compDetails.deviationThresholdNumerator)).div(fixedDenominator);
```